statannotations/Annotator.py

```diff
@@ -216,10 +218,11 @@ def annotate(self, line_offset=None, line_offset_to_group=None):
         self.validate_test_short_name()
 
         for annotation in self.annotations:
-            self._annotate_pair(annotation,
-                                ax_to_data=ax_to_data,
-                                ann_list=ann_list,
-                                orig_value_lim=orig_value_lim)
+            if self.show_non_significant or annotation.data.pvalue < 0.05:
```

Comment:
I think we should first here check that `isinstance(annotation.data, StatResult)`. Its a bit defensive but we can raise an error with a nice message to help users if they mixed that up.
Then, as the threshold is not always .05, and there can be multicomparison correction, how about using `annotation.data.pvalue <= annotation.data.alpha` if there is no correction method, or `annotation.data.corrected_significance` otherwise?

If we want to push further to increase readability, we would have a `is_significant` property on StatResult that would do this computation, and `annotate` here would simply check `annotation.data.is_significant`.
This has my preference but I could also refactor on top of your contribution.

Reply:
I like the approach you suggest (especially that there are no additional parameters to include somewhere), I'll give implementing it this way a shot !

Reply:
@sepro 
Hello Sebastian,
Can I help with anything?

Reply:
Hi, it has been a bit of a busy month and didn't have the bandwidth to tackle this. It still is on my radar and something I would like to do, but if you can get to it faster your are obviously welcome to do so as well.